## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 474585`
Commit: [`d3daad7615e8941ff585173c99259afd856d342b`](https://github.com/NixOS/nixpkgs/commit/d3daad7615e8941ff585173c99259afd856d342b) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/d3daad7615e8941ff585173c99259afd856d342b..pull/474585/head))
Merge: [`f6cce1cf67bfce7ce9f27ea0490568d6dba73a6e`](https://github.com/NixOS/nixpkgs/commit/f6cce1cf67bfce7ce9f27ea0490568d6dba73a6e)

Logs: https://github.com/Moraxyc/nixpkgs-review-gha/actions/runs/20541126410


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>adms</li>
    <li>qucs-s</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>adms</li>
    <li>qucs-s</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>qucs-s</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>adms</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>qucs-s</summary>
<pre>make[2]: *** [translations/CMakeFiles/translations.dir/build.make:161: translations/qucs_sv.qm] Segmentation fault: 11
make[1]: *** [CMakeFiles/Makefile2:1974: translations/CMakeFiles/translations.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 15%] Building CXX object qucs-s-spar-viewer/CMakeFiles/qucs-sspar-viewer.dir/codeeditor.cpp.o
[ 15%] Building CXX object qucs-s-spar-viewer/CMakeFiles/qucs-sspar-viewer.dir/smithchartwidget.cpp.o
[ 15%] Built target magnetics_autogen
[ 15%] Building CXX object qucs-s-spar-viewer/CMakeFiles/qucs-sspar-viewer.dir/rectangularplotwidget.cpp.o
[ 16%] Building CXX object qucs-s-spar-viewer/CMakeFiles/qucs-sspar-viewer.dir/polarplotwidget.cpp.o
[ 16%] Building CXX object qucs-s-spar-viewer/CMakeFiles/qucs-sspar-viewer.dir/matrixcombopopup.cpp.o
[ 16%] Building CXX object qucs-s-spar-viewer/CMakeFiles/qucs-sspar-viewer.dir/moc_qucs-s-spar-viewer.cpp.o
[ 16%] Building CXX object qucs-s-spar-viewer/CMakeFiles/qucs-sspar-viewer.dir/moc_codeeditor.cpp.o
[ 16%] Building CXX object qucs-s-spar-viewer/CMakeFiles/qucs-sspar-viewer.dir/moc_smithchartwidget.cpp.o
[ 16%] Building CXX object qucs-s-spar-viewer/CMakeFiles/qucs-sspar-viewer.dir/moc_rectangularplotwidget.cpp.o
[ 17%] Building CXX object qucs-s-spar-viewer/CMakeFiles/qucs-sspar-viewer.dir/moc_polarplotwidget.cpp.o
[ 17%] Building CXX object qucs-s-spar-viewer/CMakeFiles/qucs-sspar-viewer.dir/moc_matrixcombopopup.cpp.o
[ 17%] Building CXX object qucs-s-spar-viewer/CMakeFiles/qucs-sspar-viewer.dir/qrc_qucs-s-spar-viewer.cpp.o
Copying OS X content qucs-s-spar-viewer/qucs-sspar-viewer.app/Contents/Resources/qucs-s-spar-viewer.icns
[ 17%] Linking CXX executable qucs-sspar-viewer.app/Contents/MacOS/qucs-sspar-viewer
[ 17%] Built target qucs-sspar-viewer
make: *** [Makefile:146: all] Error 2</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>adms</li>
    <li>qucs-s</li>
  </ul>
</details>

--- a/pkgs/by-name/ad/adms/package.nix
+++ b/pkgs/by-name/ad/adms/package.nix
@@ -2,6 +2,7 @@
   lib,
   stdenv,
   fetchFromGitHub,
+  fetchpatch2,
   autoreconfHook,
   flex,
   bison,
@@ -22,6 +23,15 @@ stdenv.mkDerivation rec {
     rev = "release-${version}";
     sha256 = "0i37c9k6q1iglmzp9736rrgsnx7sw8xn3djqbbjw29zsyl3pf62c";
   };
+
+  patches = [
+    # fix build with c23
+    #   admsXml.c:645:8: error: too many arguments to function 'verilogaparse'; expected 0, have 1
+    (fetchpatch2 {
+      url = "https://salsa.debian.org/science-team/adms/-/raw/01ef4a94a48736c49c67d90da506b34f6114f0b0/debian/patches/0002-fix-ftbfs-gcc-15.patch";
+      hash = "sha256-rSNBqdpuXA9ViyygRGn4KVknLCu0Q+UoOGLfoNAgccc=";
+    })
+  ];
 
   nativeBuildInputs = [ autoreconfHook ];
   buildInputs = [

